product-backend/src/test/java/AuthServiceTest.java

```diff
@@ -97,9 +99,9 @@ void loginUser_When_UserIsValid_Expect_TokenReturned() throws SQLException, Inva
         when(passwordServiceMock.verifyHash(mockedLogin.getPassword(), existingUserMock.get().getPassword())).thenReturn(true);
         when(authDaoMock.getUserByEmail(mockedLogin.getEmail(), databaseConnectorMock.getConnection())).thenReturn(existingUserMock);
 
-        String token = authService.login(mockedLogin);
+        UserCredentials token = authService.login(mockedLogin);
 
-        assertThat(token).isNotEmpty();
+        assertThat(token).isNotNull();
```

Comment:
Update assertions
- validate token is long string
- roleId is passed to the returned value